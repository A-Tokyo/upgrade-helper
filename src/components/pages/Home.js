import React, { useState, useEffect } from 'react'
import styled from 'styled-components'
import { Card } from 'antd'
import GitHubButton from 'react-github-btn'
import ReactGA from 'react-ga'
import VersionSelector from '../common/VersionSelector'
import DiffViewer from '../common/DiffViewer'
import { homepage } from '../../../package.json'

const Page = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding-top: 30px;
`

const Container = styled(Card)`
  width: 90%;
  border-radius: 3px;
`

const TitleContainer = styled.div`
  display: flex;
  align-items: center;
`

const StarButton = styled(({ className, ...props }) => (
  <div className={className}>
    <GitHubButton {...props} />
  </div>
))`
  margin-bottom: 5px;
  margin-left: 15px;
`

const Home = () => {
  const [fromVersion, setFromVersion] = useState('')
  const [toVersion, setToVersion] = useState('')
  const [showDiff, setShowDiff] = useState(false)

  useEffect(() => {
    if (process.env.NODE_ENV === 'production') {
      ReactGA.initialize('UA-136307971-1')
      ReactGA.pageview('/')
    }
  }, [])

  const handleShowDiff = ({ fromVersion, toVersion }) => {
    if (fromVersion === toVersion) {
      return
    }

    setFromVersion(fromVersion)
    setToVersion(toVersion)
    setShowDiff(true)
  }

  return (
    <Page>
      <Container>
        <TitleContainer>
          <a href={homepage}><h1>React Native upgrade guide</h1></a>

          <StarButton
            href="https://github.com/react-native-community/upgrade-helper"
            data-icon="octicon-star"
            data-show-count="true"
            aria-label="Star react-native-community/upgrade-helper on GitHub"
          >
            Star
          </StarButton>
        </TitleContainer>

        <VersionSelector showDiff={handleShowDiff} />
      </Container>

      <DiffViewer
        showDiff={showDiff}
        fromVersion={fromVersion}
        toVersion={toVersion}
      />
    </Page>
  )
}

export default Home
