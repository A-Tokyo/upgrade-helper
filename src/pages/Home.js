import React, { useState  } from 'react'
import styled from 'styled-components'
import { Card } from 'antd'
import GitHubButton from 'react-github-btn'

import VersionSelector from '../components/common/VersionSelector'
import DiffViewer from '../components/common/DiffViewer'
import { Settings } from '../components/Settings'
import { homepage } from '../../package.json'
import logo from '../assets/logo.svg'
import { SHOW_LATEST_RCS } from '../utils'


const Page = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding-top: 30px;
`

const Container = styled(Card)`
  width: 90%;
  border-radius: 3px;
`

const TitleContainer = styled.div`
  display: flex;
  align-items: center;
`

const Subtitle = styled.div({
	display: 'flex',
	alignItems: 'center',
	marginTop: 50,
})

const LogoImg = styled.img`
  width: 100px;
  margin-bottom: 15px;
`

const TitleHeader = styled.h1`
  margin: 0;
  margin-left: 15px;
`

const StarButton = styled(({ className, ...props }) => (
	<div className={className}>
		<GitHubButton {...props} />
	</div>
))`
  margin-top: 10px;
  margin-left: 15px;
  margin-right: auto;
`

export const Home = () => {
	const [fromVersion, setFromVersion] = useState(null)
	const [toVersion, setToVersion] = useState(null)
	const [showDiff, setShowDiff] = useState(false)
	const [settings, setSettings] = useState({
		[`${SHOW_LATEST_RCS}`]: false,
	})

	const handleShowDiff = ({ fromVersion, toVersion }) => {
		if (fromVersion === toVersion) {
			return
		}

		setFromVersion(fromVersion)
		setToVersion(toVersion)
		setShowDiff(true)
	}

	const handleSettingsChange = settingsValues => {
		const normalizedIncomingSettings = settingsValues.reduce((acc, val) => {
			acc[val] = true
			return acc
		}, {})

		setSettings(normalizedIncomingSettings)
	}

	return (
		<Page>
			<Container>
				<TitleContainer>
					<LogoImg
						alt="React Native upgrade helper logo"
						title="React Native upgrade helper logo"
						src={logo}
					/>

					<a href={homepage}>
						<TitleHeader>React Native upgrade helper</TitleHeader>
					</a>

					<StarButton
						href="https://github.com/react-native-community/upgrade-helper"
						data-icon="octicon-star"
						data-show-count="true"
						aria-label="Star react-native-community/upgrade-helper on GitHub"
					>
						Star
					</StarButton>

					<Settings handleSettingsChange={handleSettingsChange} />
				</TitleContainer>

				<VersionSelector
					showDiff={handleShowDiff}
					showReleaseCandidates={settings[SHOW_LATEST_RCS]}
				/>

				<Subtitle>Don't forget: `RnDiffApp` is a placeholder. When upgrading, all instances of `RnDiffApp` should be `YourProjectName`, all `rndiffapp` should be `yourprojectname` etc.</Subtitle></Container>

			{/* <DiffViewer
				showDiff={showDiff}
				fromVersion={fromVersion}
				toVersion={toVersion}
			/> */}
		</Page>
	)
}
