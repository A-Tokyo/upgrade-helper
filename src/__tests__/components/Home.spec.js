import React from 'react'
import { render } from '@testing-library/react'
import Home from '../../components/pages/Home'

it('renders without crashing', () => {
  const { container } = render(<Home />)

  expect(container).toMatchSnapshot()
})
