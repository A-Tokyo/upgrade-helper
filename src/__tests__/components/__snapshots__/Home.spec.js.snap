// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="sc-cMljjf kjukSP"
  >
    <div
      class="ant-card sc-jAaTju jExkfr ant-card-bordered"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="sc-jDwBTQ iPrvAS"
        >
          <img
            alt="React Native upgrade helper logo"
            class="sc-gPEVay isecjX"
            src="logo.svg"
            title="React Native upgrade helper logo"
          />
          <a
            href="https://react-native-community.github.io/upgrade-helper"
          >
            <h1
              class="sc-iRbamj fwveLy"
            >
              React Native upgrade helper
            </h1>
          </a>
          <div
            class="sc-jlyJG dRjxEc"
          >
            <div>
              ReactGitHubBtn - 
              Star
            </div>
          </div>
          <button
            class="ant-btn ant-btn-icon-only"
            type="button"
          >
            <i
              aria-label="icon: setting"
              class="anticon anticon-setting"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="setting"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M924.8 625.7l-65.5-56c3.1-19 4.7-38.4 4.7-57.8s-1.6-38.8-4.7-57.8l65.5-56a32.03 32.03 0 0 0 9.3-35.2l-.9-2.6a443.74 443.74 0 0 0-79.7-137.9l-1.8-2.1a32.12 32.12 0 0 0-35.1-9.5l-81.3 28.9c-30-24.6-63.5-44-99.7-57.6l-15.7-85a32.05 32.05 0 0 0-25.8-25.7l-2.7-.5c-52.1-9.4-106.9-9.4-159 0l-2.7.5a32.05 32.05 0 0 0-25.8 25.7l-15.8 85.4a351.86 351.86 0 0 0-99 57.4l-81.9-29.1a32 32 0 0 0-35.1 9.5l-1.8 2.1a446.02 446.02 0 0 0-79.7 137.9l-.9 2.6c-4.5 12.5-.8 26.5 9.3 35.2l66.3 56.6c-3.1 18.8-4.6 38-4.6 57.1 0 19.2 1.5 38.4 4.6 57.1L99 625.5a32.03 32.03 0 0 0-9.3 35.2l.9 2.6c18.1 50.4 44.9 96.9 79.7 137.9l1.8 2.1a32.12 32.12 0 0 0 35.1 9.5l81.9-29.1c29.8 24.5 63.1 43.9 99 57.4l15.8 85.4a32.05 32.05 0 0 0 25.8 25.7l2.7.5a449.4 449.4 0 0 0 159 0l2.7-.5a32.05 32.05 0 0 0 25.8-25.7l15.7-85a350 350 0 0 0 99.7-57.6l81.3 28.9a32 32 0 0 0 35.1-9.5l1.8-2.1c34.8-41.1 61.6-87.5 79.7-137.9l.9-2.6c4.5-12.3.8-26.3-9.3-35zM788.3 465.9c2.5 15.1 3.8 30.6 3.8 46.1s-1.3 31-3.8 46.1l-6.6 40.1 74.7 63.9a370.03 370.03 0 0 1-42.6 73.6L721 702.8l-31.4 25.8c-23.9 19.6-50.5 35-79.3 45.8l-38.1 14.3-17.9 97a377.5 377.5 0 0 1-85 0l-17.9-97.2-37.8-14.5c-28.5-10.8-55-26.2-78.7-45.7l-31.4-25.9-93.4 33.2c-17-22.9-31.2-47.6-42.6-73.6l75.5-64.5-6.5-40c-2.4-14.9-3.7-30.3-3.7-45.5 0-15.3 1.2-30.6 3.7-45.5l6.5-40-75.5-64.5c11.3-26.1 25.6-50.7 42.6-73.6l93.4 33.2 31.4-25.9c23.7-19.5 50.2-34.9 78.7-45.7l37.9-14.3 17.9-97.2c28.1-3.2 56.8-3.2 85 0l17.9 97 38.1 14.3c28.7 10.8 55.4 26.2 79.3 45.8l31.4 25.8 92.8-32.9c17 22.9 31.2 47.6 42.6 73.6L781.8 426l6.5 39.9zM512 326c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 0 1 512 614c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 0 1 400 502c0-29.9 11.7-58 32.8-79.2C454 401.6 482.1 390 512 390c29.9 0 58 11.6 79.2 32.8A111.6 111.6 0 0 1 624 502c0 29.9-11.7 58-32.8 79.2z"
                />
              </svg>
            </i>
          </button>
        </div>
        <div
          class="sc-htpNat hqmgw"
        >
          <div
            class="sc-bdVaJa UoSJW"
          >
            <h4>
              What's your current React Native version?
            </h4>
            <div
              class="ant-select-lg sc-bxivhb iYdHlP sc-bwzfXH jwfMIB ant-select ant-select-enabled ant-select-loading"
            >
              <div
                aria-autocomplete="list"
                aria-controls="test-uuid"
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-select-selection
            ant-select-selection--single"
                role="combobox"
                tabindex="0"
              >
                <div
                  class="ant-select-selection__rendered"
                >
                  <div
                    class="ant-select-selection-selected-value"
                    style="display: block; opacity: 1;"
                    title=""
                  />
                </div>
                <span
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <i
                    aria-label="icon: loading"
                    class="anticon anticon-loading"
                  >
                    <svg
                      aria-hidden="true"
                      class="anticon-spin"
                      data-icon="loading"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
          </div>
          <div
            class="sc-bdVaJa UoSJW"
          >
            <h4>
              To which version would you like to upgrade?
            </h4>
            <div
              class="ant-select-lg sc-ifAKCX dTveGH sc-bwzfXH jwfMIB ant-select ant-select-enabled ant-select-loading"
            >
              <div
                aria-autocomplete="list"
                aria-controls="test-uuid"
                aria-expanded="false"
                aria-haspopup="true"
                class="ant-select-selection
            ant-select-selection--single"
                role="combobox"
                tabindex="0"
              >
                <div
                  class="ant-select-selection__rendered"
                >
                  <div
                    class="ant-select-selection-selected-value"
                    style="display: block; opacity: 1;"
                    title=""
                  />
                </div>
                <span
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <i
                    aria-label="icon: loading"
                    class="anticon anticon-loading"
                  >
                    <svg
                      aria-hidden="true"
                      class="anticon-spin"
                      data-icon="loading"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
                      />
                    </svg>
                  </i>
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-EHOje jWoLok"
        >
          <button
            class="ant-btn ant-btn-primary ant-btn-lg"
            type="button"
          >
            <span>
              Show me how to upgrade!
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
